### DISCUSSION TITLE
Build & Deploy partial files !!!

### DISCUSSION THREAD

**Discussion by Tekkasoft** [SuessLabs/VsLinuxDebug#35] [Q&A] [answered]
Hello
it is the first time that I write but I have been following the community for a long time.
I would like to start by thanking the authors of the VS extension for testing code on Linux and posing a problem that I have not been able to solve, yet.
I have a blazor serve side solution, under windows 10, in a given folder under TFS control.
The same solution, in another folder, without TFS checking.

If I do Build & Deploy, or Build & Deploy & Debug, on the second solution, it is published and launched correctly.
The same command, on the first solution, **publishes only a part of the solution**.
In particular, the **_content** folder under **wwwroot** is not published and obviously the application does not work correctly.

Apart from TFS there does not seem to be any difference between the files of the two solutions as well as between the files of the main project.
Surely I'm missing something about how to publish the whole solution, always or others setting.
Looking forward to some suggestions
Thanks in advance

**Comment by DamianSuess** [comment, marked as answer]
Hi @Tekkasoft,
Thank you for giving the extension a try, I believe I understand your enhancement request. In your scenario, there are multiple projects in the solution which you would like for them to be uploaded to the remote machine, correct?

Currently, this extension was built for simple single-project solutions and multi-projects that share the same output folder (_not, "ProjA\Bin", "ProjB\Bin"). Overall, it uploads whatever is in your startup project's output folder.

To assist in this further could you help give an example structure of your solution?  For example, one product I work on is complex and contains, UI-client, webserver, state-machine, and simulators.  Since this uses a multi-project startup and each project has its own output folder, it needs to be uploaded to the remote system in parts - _something to consider as a future enhancement_.

**Reply by Tekkasoft** [reply]
Hi Damian

the solution is multi-project which, shares the same output folder and this works well. All DLLs are copied correctly. The problem is the only folder "_content" under "wwwroot" (remember that the application is blazor server side) that is created when you publish the solution (see image annex).
![_contentFolder](https://user-images.githubusercontent.com/82211952/177956336-a6938724-ea37-402f-b922-ebd8c5500909.PNG)

Now why does the copy of the solution, in a different path from the original one, generate this folder during the Build & Deploy and when I run it for the original solution it doesn't work?
 
Could the solution path affect the behavior of the extension?

Greetings

thanks

**Reply by DamianSuess** [reply]
@tekkasoft, I tried to reproduce your issue but I've been unable to with both my company's ASP.NET WebApi projects. I even tried creating a new Blazor Server App and `Extensions > Linux Debugger > Build & Deploy` both when it was not under source control (git) and then added it to source control. Unfortunately, I don't get the same results as yours.

For the test app, I did the following
1. Created a simple "Blazor Server App" called, "BlazorTestApp1" (_the sample template's weather app_)
2. Using .NET 6.0, Auth Type: "none", [checked] Configure for HTTPS, [unchecked] Enable Docker
3. Built the app in DEBUG
4. `Extensions > Linux Debugger > Build & Deploy` 
5. Built the proj in RELEASE &  performed `Build & Deploy`
6. Added project's folder to Git source control
7. Repeated steps 3, 4, and 5.

The output folder is quite simple in each of my cases, where it's populated only with `.exe, .dll, .json, .pdb` files.

Next, I even tried performing a _Publish_, which created the `wwwroot` subfolder with only `css` folder & the usual .css & .ico files.  Didn't get a `_content` folder.

Is there something else I am missing?

**Reply by Tekkasoft** [reply]
hi Damian

thanks for the patience and attention you are dedicating to the problem. I will try to do what you say to try to reproduce the problem with other solutions as well.
As soon as I am able to reproduce the problem I will keep you updated

Greetings
